### PR TITLE
Added SKOS reader

### PR DESCRIPTION
**Pull request by arunverma87**
PR related to #165

### CODE REVIEW

File: io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/data/test1.rdf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:skos="http://www.w3.org/2004/02/skos/core#" xmlns:dc="http://purl.org/dc/elements/1.1/">
+	<skos:Concept rdf:about="http://vocab.to.wetransform/term/W080/1/CONT0006">
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/data/test1.rdf]: Is that correct, concept with an upper C? Did you check the respective schema? [Here](http://vocab.ices.dk/services/rdf/collection/PARAM) for instance a lower case element is used.

File: io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/eu/esdihumboldt/hale/io/codelist/skos/test/SkosCodeListTest.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	 * 
+	 * test url 2 consisting concepts (case: fallback).
+	 * 
+	 * @throws Exception throws exception if something wrong occurs
+	 */
+	@Test
+	public void testSKOSFromURL2() throws Exception {
+
+		CodeList codeList = readCodeList(
+				URI.create("http://vocab.ices.dk/services/rdf/collection/PARAM"));
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/eu/esdihumboldt/hale/io/codelist/skos/test/SkosCodeListTest.java]: I don't think really relying on the online resource for that test makes sense. The online resource may change and thus the number of concepts or their definition.

File: io/plugins/eu.esdihumboldt.hale.io.codelist.skos/plugin.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?eclipse version="3.4"?>
+<plugin>
+   <extension
+         point="org.eclipse.core.contenttype.contentTypes">
+      <content-type
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos/plugin.xml]: The XML content type could be used as parent content type.

File: io/plugins/eu.esdihumboldt.hale.io.codelist.skos/src/eu/esdihumboldt/hale/io/codelist/skos/reader/SkosCodeListReader.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
+			throws IOProviderConfigurationException, IOException {
+		progress.begin("Loading SKOS code list", ProgressIndicator.UNKNOWN);
+		try {
+			URI loc = getSource().getLocation();
+			InputStream in = getSource().getInput();
+			codelist = new SkosCodeList(in, loc);
+			progress.setCurrentTask("Code list loaded.");
+		} catch (Exception ex) {
+			throw new RuntimeException(ex);
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos/src/eu/esdihumboldt/hale/io/codelist/skos/reader/SkosCodeListReader.java]: You should handle the exception here, report the error and report the whole process as failed.

File: io/plugins/eu.esdihumboldt.hale.io.codelist.skos/src/eu/esdihumboldt/hale/io/codelist/skos/reader/SkosCodeList.java [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+	public CodeEntry getEntryByIdentifier(String identifier) {
+		return entriesByIdentifier.get(identifier);
+	}
+
+	@Override
+	public URI getLocation() {
+		return location;
+	}
+
+	private boolean getConceptSchemes() {
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos/src/eu/esdihumboldt/hale/io/codelist/skos/reader/SkosCodeList.java]: All these get(ter) methods w/ boolean or void return type confuse me. Usually this naming convention is used for getters, but these seem to be methods that mostly are there for their side effects.

File: io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/eu/esdihumboldt/hale/io/codelist/skos/test/SkosCodeListTest.java (hunk 2 above)
Comment (arunverma87) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/eu/esdihumboldt/hale/io/codelist/skos/test/SkosCodeListTest.java]: I want to test the fallback scenario, which is happening in this link only ( skos api could not load it). So i added this online resource.

File: io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/data/test1.rdf (hunk 1 above)
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/data/test1.rdf]: Ok, interesting. Is this then actually loaded by the SKOS API or by the XML mechanism? I think I saw in the XML mechanism that it uses lower case "concept".

Comment (arunverma87) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/data/test1.rdf]: It is loaded by skos API itself.

Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/data/test1.rdf]: Ok, so it is not unlikely that the usage of `skos:concept` in other places is incorrect, and we only support it for compatibility reasons.

**Pull request merged**